capturebody.go

```diff
@@ -148,7 +155,7 @@ func (bc *BodyCapturer) setContext(out *model.RequestBody) bool {
 		return true
 	}
 
-	body, n := apmstrings.Truncate(bc.buffer.String(), stringLengthLimit)
+	body, n := bc.getBufferTruncated(bc.mu.RLocker())
```

Comment:
I'm not sure what my thought process was there, but something like: the caller might want to get the buffer with a write lock instead of a read lock. The code doesn't do that though. I'll change it to call `bc.mu.Rlock` internally.